### PR TITLE
writers: support for more writers

### PR DESCRIPTION
**Pull request by tulilirockz**
## Description of changes

This PR adds some functions to `pkgs.writers`, those being: `write{Nu,Ruby,Lua}{,Bin}` so that people can have support for more useful ""glue"" languages for their nix scripts.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/build-support/writers/scripts.nix [hunk 1]
```diff
@@ -185,6 +185,46 @@ rec {
...
+    writeRuby "/bin/${name}";
+
+  # Like writeScript but the first line is a shebang to lua
+  #
+  # Example:
+  #   writeLua "example" ''
+  #    print("hello world")
+  #   ''
+  writeLua = makeScriptWriter {
+    interpreter = "${pkgs.lua}/bin/lua";
```
Comment (SuperSandro2000) [inline pkgs/build-support/writers/scripts.nix:222]: We could use lib.getExe here and for the other interpreters

Comment (tulilirockz) [reply pkgs/build-support/writers/scripts.nix:222]: True! Do you want me to add this to all the interpreters and try to test them out?

**Comment by Lassulus** [comment]
Thanks for adding more writers! Would be nice to have some tests so we can detect breakage easier. Should be fairly straightforward in the tests.nix. if you have more questions just ask me

**Comment by tulilirockz** [comment]
Sorry, but I've been messing around and haven't found a way to run that `test.nix` file by itself, how could I do so? `nix-instantiate --eval` doesnt work I think... So I don't know

**Comment by Lassulus** [comment]
ah you can run the test like this in a checkout of nixpkgs:
`nix-build . -A tests.writers`

although it seems like the pypy and fsharp tests are broken right now. maybe we should comment them out. also a comment (somwhere in the code) how to run the writers tests could be helpful :)

**Comment by tulilirockz** [comment]
Thank you very much for the help! I was doing it backwards, running `nix-build -A writers.tests`, lol! So, apparently the pypy tests are broken because of [this issue](https://github.com/NixOS/nixpkgs/issues/39356) so they should be commented out for now indeed!

**Comment by Lassulus** [comment]
I would comment out the tests and add a link to the issue next to it? I guess we can let the writers still exist although they are not working with libraries right now.
By exposing the interface there is a higher chance than someone stumbles upon it and finds time/motivation to fix it :)

**Comment by tulilirockz** [comment]
I think that's pretty much all I wanted to add in this PR! The tests are now working, sadly Lua and Ruby packages don't work, though :(. I think this PR is pretty much done at this point.

**Comment by Lassulus** [comment]
would be nice if the commit messages could follow the https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions . maybe we should also get rid of the master merge commit

**Comment by Lassulus** [comment]
I can also amend the history if that's okay for you? Then we can merge it

**Comment by tulilirockz** [comment]
Yup! no prob sorry for not answering before

**Pull request merged**